controllers/restic.go

```diff
@@ -82,40 +82,39 @@ func (r *VolumeSnapshotBackupReconciler) CreateVSBResticSecret(log logr.Logger)
 	var scheduleCronExpr = ""
 	rpolicy := RetainPolicy{}
 	for key, val := range resticSecret.Data {
+		stringVal := string(val)
 		if key == ResticRepository {
 			// if trailing '/' in user-created Restic repo, remove it
-			stringVal := string(val)
-			stringVal = strings.TrimRight(stringVal, "/")
-
-			ResticRepoValue = stringVal
+			ResticRepoValue = strings.TrimRight(stringVal, "/")
 		}
 		if key == ResticPruneInterval {
-			pruneInterval = string(val)
+			pruneInterval = stringVal
 		}
 		if key == SnapshotScheduleCron {
-			scheduleCronExpr = string(val)
+			scheduleCronExpr = stringVal
 		}
 		if key == SnapshotRetainPolicyMonthly {
-			rpolicy.monthly = string(val)
+			rpolicy.monthly = stringVal
 		}
 		if key == SnapshotRetainPolicyDaily {
-			rpolicy.daily = string(val)
+			rpolicy.daily = stringVal
 		}
 		if key == SnapshotRetainPolicyHourly {
-			rpolicy.hourly = string(val)
+			rpolicy.hourly = stringVal
 		}
 		if key == SnapshotRetainPolicyWeekly {
-			rpolicy.weekly = string(val)
+			rpolicy.weekly = stringVal
 		}
 		if key == SnapshotRetainPolicyYearly {
-			rpolicy.yearly = string(val)
+			rpolicy.yearly = stringVal
 		}
 		if key == SnapshotRetainPolicyWithin {
-			rpolicy.within = string(val)
+			rpolicy.within = stringVal
 		}
 	}
 
-	resticrepo := fmt.Sprintf("%s/%s/%s", ResticRepoValue, pvc.Namespace, pvc.Name)
+	// TODO should check if label is present first? or it is always present?
+	resticrepo := fmt.Sprintf("%s/%s/%s/%s", ResticRepoValue, pvc.Namespace, vsb.Labels["velero.io/backup-name"], pvc.Name)
```

Comment:
the label key is defined in vsm-plugin code https://github.com/migtools/velero-plugin-for-vsm/blob/main/internal/util/labels_annotations.go#L53

Should we use it instead of putting the string here? the problem is that is vsm-plugin that uses vsm-controller, so we would have to define a variable with the label key here and then import in plugin code, right?

But I think I heard that the idea is to use a native vsm in future OADP, so maybe this change is not necessary